metricbeat/module/kubernetes/state_container/_meta/fields.yml

```diff
@@ -0,0 +1,49 @@
+- name: container
+  type: group
+  description: >
+    kubernetes container metrics
+  fields:
+    - name: resource
+      type: group
+      description: >
+        Kubernetes container resource requests and limits
+      fields:
+        - name: status
+          type: group
+          fields:
+            - name: phase
+              type: keyword
+              description: >
+                Container phase (running, waiting, terminated)
+            - name: ready
+              type: boolean
+              description: >
+                Container ready status
+            - name: restarts
+              type: integer
+              description: >
+                Container restarts count
+        - name: limits
+          type: group
+          fields:
+            - name: memory.bytes
+              type: long
+              format: bytes
+              description: >
+                Container memory usage limit in bytes
+            - name: cpu.nanocores
+              type: long
+              description: >
+                Container CPU nanocores limit
+        - name: requested
+          type: group
+          fields:
+            - name: memory.bytes
+              type: long
+              format: bytes
+              description: >
+                Container requested memory in bytes
+            - name: cpu.nanocores
+              type: long
+              description: >
+                Container CPU requested nanocores
```

Comment:
A common scenario is to check requested vs limits vs real usage. Ideally you would put some sane limits based on real usage of a given container